src/V3Timing.cpp

```diff
@@ -0,0 +1,785 @@
+        // Handle the intra assignment timing control (or transform if we're in a suspendable)
+        // Create an intermediate var
+        auto* const newvscp
+            = m_scopep->createTemp(m_dlyVarNames.get(nodep->lhsp()), nodep->lhsp()->width());
+        // Add a bool var that says if we want to commit this NBA
+        auto* const chgVscp = m_scopep->createTemp(m_dlyChgVarNames.get(nodep), 1);
+        m_procp->addNextHere(new AstInitialStatic{
+            flp, new AstAssign{flp, new AstVarRef{flp, chgVscp, VAccess::WRITE},
+                               new AstConst{flp, AstConst::BitFalse()}}});
+        // Create an AlwaysPost that will be responsible for committing it
+        auto* const ifp = new AstIf{flp, new AstVarRef{flp, chgVscp, VAccess::READ},
+                                    new AstAssignDly{flp, nodep->lhsp()->unlinkFrBack(),
+                                                     new AstVarRef{flp, newvscp, VAccess::READ}}};
+        ifp->addIfsp(new AstAssign{flp, new AstVarRef{flp, chgVscp, VAccess::WRITE},
+                                   new AstConst{flp, AstConst::BitFalse()}});
+        auto* const alwaysp = new AstAlwaysPost{
+            flp,
+            new AstSenTree{flp, new AstSenItem{flp, VEdgeType::ET_CHANGED,
+                                               new AstVarRef{flp, chgVscp, VAccess::READ}}},
+            ifp};
+        m_procp->addNext(alwaysp);
+        // Signal that we are doing this assignment
+        auto* const enablep = new AstAssign{flp, new AstVarRef{flp, chgVscp, VAccess::WRITE},
+                                            new AstConst{flp, AstConst::BitTrue()}};
+        nodep->replaceWith(enablep);
+        // Assign to the temporary var
+        auto* const assignp = new AstAssign{flp, new AstVarRef{flp, newvscp, VAccess::WRITE},
+                                            nodep->rhsp()->unlinkFrBack()};
+        if (controlp) {
+            if (m_underFork) {
+                controlp->replaceWith(new AstBegin{flp, "", assignp});
+                assignp->addNextHere(controlp);  // No need to create another fork
+            } else {
+                controlp->replaceWith(assignp);
+                auto* const forkp = new AstFork{flp, "", controlp};
+                forkp->joinType(VJoinType::JOIN_NONE);
+                assignp->addNextHere(forkp);
+            }
+        } else {
+            enablep->addNextHere(assignp);
+        }
+        nodep->deleteTree();
+    }
+    virtual void visit(AstWait* nodep) override {
+        // Gather all vars we want to wait on
+        VarScopeSet waitVars;
+        nodep->foreach<AstVarRef>([&](AstVarRef* refp) { waitVars.insert(refp->varScopep()); });
+        if (waitVars.empty()) {  // There are no vars to wait on
+            if (nodep->bodysp())
+                nodep->replaceWith(nodep->bodysp()->unlinkFrBackWithNext());
+            else
+                nodep->unlinkFrBack();
```

Comment:
what if it was `wait (false)` ?

Reply:
And add a test for wait(false) as seems a likely bug farm.

Also use "{} else {}" when multiline.